pandas/core/dtypes/dtypes.py

```diff
@@ -550,8 +550,15 @@ def update_dtype(self, dtype):
             new_categories = self.categories
 
         new_ordered = dtype.ordered
+
+        # TODO(GH26336): remove this if block when ordered=None is removed
         if new_ordered is None:
             new_ordered = self.ordered
+            if self.ordered:
+                msg = ("ordered=None is deprecated and will default to False "
+                       "in a future version; ordered=True must be explicitly "
+                       "passed in order to be retained")
+                warnings.warn(msg, FutureWarning, stacklevel=2)
```

Comment:
Maybe we should let the stacklevel fit for the astype case (although the series and index case might need a different stacklevel), as that seems the more common case compared to directly using this method? 

Or, otherwise, I would certainly try to make the context clearer in this warning message: indicate that a CategoricalDtype was constructed without specifying the ordered, etc (otherwise the message might be very confusing, as it is not raised when creating it)

Reply:
I've updated the `stacklevel` so that it works for the `Categorical.astype` and `CategoricalIndex.astype` cases; all other cases look like they require a unique `stacklevel`.  I've also updated the message itself to be more clear.